src/webgpu/api/validation/non_filterable_texture.spec.ts

```diff
@@ -0,0 +1,127 @@
+export const description = `
+Tests that non-filterable textures used with filtering samplers generate a validation error.
+`;
+
+import { makeTestGroup } from '../../../common/framework/test_group.js';
+import { keysOf } from '../../../common/util/data_tables.js';
+
+import { ValidationTest } from './validation_test.js';
+
+const kNonFilterableCaseInfo: {
+  [key: string]: { sampleType: GPUTextureSampleType; component: string };
+} = {
+  i32: { sampleType: 'sint', component: '0,' },
+  u32: { sampleType: 'uint', component: '0,' },
+  f32: { sampleType: 'float', component: '0,' }, // no error for f32
```

Comment:
what about `'unfilterable-float'`?
If intentionally skipping it, just leave a short comment about why (maybe because it's already tested elsewhere, though probably no harm in testing it here too)

Reply:
added